baremetal/metal3machine_manager.go

```diff
@@ -1741,3 +1791,83 @@ func (m *MachineManager) getNodesWithLabel(ctx context.Context, nodeLabel string
 	}
 	return nodes, nodesCount, err
 }
+
+// getNodesWithoutLabel gets kubernetes nodes based on their Spec.providerID field.
+func (m *MachineManager) getNodesWithoutLabel(ctx context.Context, providerIDLegacy, providerIDNew string, providerIDonM3M *string, clientFactory ClientGetter) (int, error) {
```

Comment:
should we rename this into something else, since we are not getting nodes without labels, but more of a counting matching nodes? maybe to `getMatchingNodesCount` or something similar

Reply:
I agree. 
I was trying to emphasise  that fact that labels are not in user. I should probably rename it and add comments about the label.

Reply:
Changed it to `getMatchingNodesWithoutLabelCount`